src/sentry/static/sentry/app/views/eventsV2/eventView.tsx

```diff
@@ -662,7 +676,7 @@ class EventView {
       field: string;
       fieldname: string;
     },
-    tableDataMeta: MetaType
+    tableDataMeta: MetaType | undefined
```

Comment:
If we used an empty/null implementation of the MetaType would we be able to avoid some of the type casting? The `getSortKeyFromField` method looks like it could be adapted to handle missing keys in `MetaType` allowing us to combine `getSortKeyFromField()` and `getSortKeyFromFieldWithoutMeta()` as they are the same basic logic.

Reply:
@markstory I've refactored out `getSortKeyFromFieldWithoutMeta()`